akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala

```diff
@@ -455,13 +456,29 @@ final case class `Content-Location`(uri: Uri) extends jm.headers.ContentLocation
  * two there is slight but important difference regarding how parameter values are formatted. In RFC6266 parameters values are without quotes and
  * in RFC2616 they are quoted. Since common practice among http servers is to understand quoted values, we use older document
  * as reference here.
+ * Extended (i.e. encoded) parameter values are not quoted, see ext-parameter - https://tools.ietf.org/html/rfc5987#section-3.2.1
  */
 object `Content-Disposition` extends ModeledCompanion[`Content-Disposition`]
 final case class `Content-Disposition`(dispositionType: ContentDispositionType, params: Map[String, String] = Map.empty)
   extends jm.headers.ContentDisposition with RequestResponseHeader {
   def renderValue[R <: Rendering](r: R): r.type = {
+    import Rendering.{ contentDispositionFilenameSafeChars => safeChars }
     r ~~ dispositionType
-    params foreach { case (k, v) => r ~~ "; " ~~ k ~~ '=' ~~#! v }
+    val renderExtFilename =
+      params.get("filename").map(safeChars.indexOfFirstMismatch(_) >= 0) getOrElse false
```

Comment:
Can this be rewritten as something like `params.get("filename").exists(safeChars(_))`?

Reply:
.exists(!safeChars(_)) ?

Reply:
Ah, yes, indeed :)